### PR TITLE
feature(likes): Delete permissions of likes for like owners, liked entity owners and group owners

### PR DESCRIPTION
**Pull request by iionly**

### CODE REVIEW

File: mod/likes/start.php [hunk 1]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+
 	if (!$annotation || $annotation->name !== 'likes') {
 		return $return;
 	}
-	
-	$owner = $annotation->getOwnerEntity();
-	if (!$owner) {
-		return $return;
+
+	if ($annotation->owner_guid == $user->guid) {
```
Comment (hypeJunction) [inline mod/likes/start.php]: Is this necessary? I think it's the default

File: mod/likes/start.php [hunk 2]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+	    return true;
 	}
-	
-	return $owner->canEdit();
+
+	$liked_on = get_entity($annotation->entity_guid);
+	if ($liked_on->owner_guid == $user->guid) {
```
Comment (hypeJunction) [inline mod/likes/start.php]: Is this necessary? I think it's the default.

File: mod/likes/start.php [hunk 3]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+	    return true;
+	}
+
+	$container = $liked_on->getContainerEntity();
+	if (elgg_instanceof($container, 'group') && $container->owner_guid == $user->guid) {
```
Comment (hypeJunction) [inline mod/likes/start.php]: `$container->canEdit()`

File: mod/likes/start.php [hunk 4]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+		return true;
+	} else if ((elgg_instanceof($liked_on, 'object', 'comment') || elgg_instanceof($liked_on, 'object', 'discussion_reply')) && elgg_instanceof($container, 'object')) {
```
Comment (hypeJunction) [inline mod/likes/start.php]: `instanceof ElggComment`

File: mod/likes/start.php [hunk 5]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+		$container = $container->getContainerEntity();
+		if (elgg_instanceof($container, 'group') && $container->owner_guid == $user->guid) {
```
Comment (hypeJunction) [inline mod/likes/start.php]: `$container->canEdit()`

File: mod/likes/start.php [hunk 6]
```diff
@@ -38,18 +38,33 @@ function likes_init() {
...
+			return true;
+		}
+	}
+
+	return false;
```
Comment (hypeJunction) [inline mod/likes/start.php]: I'd rather not. We should work from reverse.

**Review by hypeJunction — CHANGES_REQUESTED** [review]

**Comment by iionly** [comment]
Closing. Will provide this functionality with a plugin instead.

**Pull request closed without merging**